devicetypes/smartthings/ezex-smart-electric-switch.src/ezex-smart-electric-switch.groovy

```diff
@@ -56,13 +58,24 @@ def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
 def parse(String description) {
     log.debug "description is $description"
     def event = zigbee.getEvent(description)
+    def powerDiv = 0
+    def energyDiv = 0
+
+    if (device.getDataValue("model") == "ZB-ONOFFPlug-D0000" || device.getDataValue("model") == "ZB-ONOFFPlug-D0005") {
```

Comment:
this conditional should be broken out into its own function. And I think you'd be okay just using

`device.getDataValue("manufacturer") == "LDS"`

You could also save some lines of code by setting `powerDiv` and `energyDiv` to the default values first and only changing them if the manufacturer check is true.